src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/fs/TeeStepTest.java

```diff
@@ -74,12 +76,55 @@ public void smokes() throws Exception {
         rr.then(r -> {
                 SemaphoreStep.success("wait/1", null);
                 WorkflowRun b = r.jenkins.getItemByFullName("p", WorkflowJob.class).getBuildByNumber(1);
-                r.waitForCompletion(b);
+                r.assertBuildStatus(Result.SUCCESS, r.waitForCompletion(b));
                 assertThat(r.getLog(b), stringContainsInOrder("got: first message second message", Functions.isWindows() ? "WS>rem" : "+ true"));
 
         });
     }
 
+    @Test
+    @Issue({"JENKINS-54346", "JENKINS-55505"})
+    public void closed() throws Exception {
+        rr.then(r -> {
```

Comment:
Still no test coverage of behavior across restarts that I can tell.

Reply:
See my last commit.
Is that what you expect?

Reply:
Something like that, yes. I would have to immerse myself in this patch for a while to really remember what the tricky cases were.